tests/test_ptc.py

```diff
@@ -253,6 +258,30 @@ def test_covAstier(self):
                     extractPtc.covariancesSqrtWeights[ampName][0],
                     ptc.covariancesSqrtWeights[ampName][i],
                 )
```

Comment:
`assertFloatsAlmostEqual` can work with arrays, this can be just one call for each; or maybe you're only checking the diagonals?  Wait, this is a dict ...  that isn't correct.

```
noiseMatrixExpected = np.array(
    [[29.37, 0, 0],
     [0, -23.209, 0],
     [0, 0, 35.69]]
)
```
above (or maybe do the full matrix with off-diagonals?) would certainly be a cleaner comparison (and also for the noB case).

Reply:
Thanks. I now use the full matrix too.